Uplift/Utils/Extensions/Array+Extension.swift

```diff
@@ -9,6 +9,24 @@
 import Foundation
 import UpliftAPI
 
+extension Array where Element == Equipment {
+
+    /// Map an array of `EquipmentFields` to an array of `Equipment` objects.
+    init(_ equipments: [EquipmentFields]) {
+        self.init(equipments.map(Equipment.init))
+    }
+
+    /// Returns all equipment types in this array.
+    func allTypes() -> [EquipmentType] {
+        var unique = [EquipmentType]()
+        for eqmt in self where !unique.contains(where: { $0 == eqmt.equipmentType }) {
+            unique.append(eqmt.equipmentType)
+        }
+        return unique
```

Comment:
```suggestion
        self.compactMap(\.equipmentType)
```
We can simplify your logic to one line. Note that we are using `.compactMap` because `equipmentType` is now an optional.